### PR TITLE
chore: 🤖 fix eslint and build issues

### PR DESCRIPTION
**Pull request by BolajiAyodeji**
#### What does this PR do

This PR fixes the issue with the production build failing when installing next `12.1.0` and conflicting dependencies between the use of npm and yarn (`error next@12.1.0: The engine "node" is incompatible with this module. Expected version ">=12.22.0". Got "12.18.0"`). I've added the required node version to Netlify and re-ran the build internally (so it passes now) and the `.nvmrc` file (for optional `nvm` usage in development).

Previously Eslint wasn't running in production, and this PR fixes that alongside some adjustments to fix all breaking rules.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -27,7 +27,6 @@
           "afterColon": true
         }
       ],
-      "react/require-default-props": [2, { "forbidDefaultForRequired": true }],
```
Comment (Princesso) [inline .eslintrc]: Any reason we do not want components to have default props?

**Review by Princesso — COMMENTED** [review]
Is there a reason why we want to remove the check for default props? Once there's a response, I'll approve. Everything LGTM

File: .eslintrc (hunk 1 above)
Comment (BolajiAyodeji) [reply .eslintrc]: I wasn't involved with the current setup, so I don't know the reason behind some decisions, but I observed that the`className` prop has an optional string type. In some components, the className props and type is defined but not used in that file, hence having default props would just be a redundant className = null or so.

Maybe the best thing, then, is to remove the props and types in files that wouldn't use need it and make it required in files that need it? I didn't think of this. I'll revert accordingly.

Comment (BolajiAyodeji) [reply .eslintrc]: The same issue applies to so many other props, actually.

Comment (BolajiAyodeji) [reply .eslintrc]: I guess the issue, then, is that in some files, the component is reused twice, with one requiring the props and the other not. Hence the props type cannot be optional nor required.

**Comment by BolajiAyodeji** [comment]
You can review again @Princesso; thank you.

**Review by Princesso — APPROVED** [review]

**Pull request merged**